src/Streamly/Internal/Data/Fold.hs

```diff
@@ -2193,3 +2197,36 @@ toParallelSVarLimited svar winfo = Fold step initial extract
 
     extract True = liftIO $ sendStop svar winfo
     extract False = return ()
+
+------------------------------------------------------------------------------
+-- Deprecated
+------------------------------------------------------------------------------
+
+-- | Flatten the monadic output of a fold to pure output.
+--
+-- @since 0.7.0
+{-# DEPRECATED sequence "Use rsequence instead" #-}
+{-# INLINE sequence #-}
+sequence :: Monad m => Fold m a (m b) -> Fold m a b
+sequence (Fold step initial extract) = Fold step' initial1 extract'
+
+    where
+
+    eval res =
+        case res of
+            Partial x -> return $ Partial x
+            Done b -> Done <$> b
+
+    initial1 = initial >>= eval
+
+    step' s a = step s a >>= eval
+
+    extract' = join . extract
+
+-- | Map a monadic function on the output of a fold.
+--
+-- @since 0.7.0
+{-# DEPRECATED mapM "Use rmapM instead" #-}
+{-# INLINE mapM #-}
+mapM :: Monad m => (b -> m c) -> Fold m a b -> Fold m a c
+mapM f = sequence . fmap f
```

Comment:
This should be defined as `mapM = rmapM`. Also, we do not need a separate section for deprecated functions. They can be defined next to the main definitions. Similar functions should be together deprecated or not.